presto-main/src/test/java/com/facebook/presto/sql/query/TestUnnest.java

```diff
@@ -13,23 +13,116 @@
  */
 package com.facebook.presto.sql.query;
 
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 public class TestUnnest
 {
+    private QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
     @Test
     public void testUnnestArrayRows()
     {
-        try (QueryAssertions assertions = new QueryAssertions()) {
-            assertions.assertQuery(
-                    "SELECT * FROM UNNEST(ARRAY[ROW(1, 1.1), ROW(3, 3.3)], ARRAY[ROW('a', true), ROW('b', false)])",
-                    "VALUES (1, 1.1, 'a', true), (3, 3.3, 'b', false)");
-            assertions.assertQuery(
-                    "SELECT x, y FROM (VALUES (ARRAY[ROW(1.0, 2), ROW(3, 4.123)])) AS t(a) CROSS JOIN UNNEST(a) t(x, y)",
-                    "VALUES (1.0, 2), (3, 4.123)");
-            assertions.assertQuery(
-                    "SELECT x, y, z FROM (VALUES (ARRAY[ROW(1, 2), ROW(3, 4)])) t(a) CROSS JOIN (VALUES (1), (2)) s(z) CROSS JOIN UNNEST(a) t(x, y)",
-                    "VALUES (1, 2, 1), (1, 2, 2), (3, 4, 1), (3, 4, 2)");
-        }
+        assertions.assertQuery(
+                "SELECT * FROM UNNEST(ARRAY[ROW(1, 1.1), ROW(3, 3.3)], ARRAY[ROW('a', true), ROW('b', false)])",
+                "VALUES (1, 1.1, 'a', true), (3, 3.3, 'b', false)");
+        assertions.assertQuery(
+                "SELECT x, y FROM (VALUES (ARRAY[ROW(1.0, 2), ROW(3, 4.123)])) AS t(a) CROSS JOIN UNNEST(a) t(x, y)",
+                "VALUES (1.0, 2), (3, 4.123)");
+        assertions.assertQuery(
+                "SELECT x, y, z FROM (VALUES (ARRAY[ROW(1, 2), ROW(3, 4)])) t(a) CROSS JOIN (VALUES (1), (2)) s(z) CROSS JOIN UNNEST(a) t(x, y)",
+                "VALUES (1, 2, 1), (1, 2, 2), (3, 4, 1), (3, 4, 2)");
+    }
+
+    @Test
+    public void testUnnestPreserveColumnName()
+    {
+        assertions.assertQuery(
+                "SELECT x FROM UNNEST(array[" +
```

Comment:
It's not immediately obvious that `array(cast(...) as row(x int, y varchar), cast(...) as row (x int, y varchar)))` is of type 
 `array(row(x int, y varchar))`.

A better form would be: `cast(array[row(1, 'a'), row(2, 'b')] as array(row(x int, y varchar)))`